src/index.ts

```diff
@@ -1790,6 +1794,7 @@ export class GossipSub extends EventEmitter<GossipsubEvents> implements PubSub<G
     this.metrics?.onJoin(topic)
 
     const toAdd = new Set<PeerIdStr>()
+    const backoff = this.backoff.get(topic) || new Map()
```

Comment:
Do not allocate a new map if not necessary, handle the undefined value case below